src/test/java/uk/nhs/prm/deduction/e2e/tests/RepositoryE2ETests.java

```diff
@@ -112,6 +150,134 @@ void shouldIdentifyEhrRequestAsEhrOutMessage() {
+            String idRootValue = node.getAttributes().getNamedItem("root").getNodeValue();
+            // return false to skip comparison in case when id root value itself is a message id
+            if (isValidUUID(idRootValue)) {
+                return false;
+            }
+            // return false to skip comparison when the type of compared value is in the excludedList
+            return !(node.getNodeName().equals("id") && excludeList.contains(idRootValue));
+        }
+        return true;
+    }
+
+    @Test
+    public void given2XMLsWithDifferences_whenTestsSimilarWithDifferenceEvaluator_thenCorrect() {
+        // helper test case for building the `excludeComparisons` function
+//        final String control = "<agentOrganizationSDS classCode=\"ORG\" determinerCode=\"INSTANCE\"><id root=\"1.2.826.0.1285.0.1.10\" extension=\"B85002\"/></agentOrganizationSDS>";
+//        final String test =    "<agentOrganizationSDS classCode=\"ORG\" determinerCode=\"INSTANCE\"><id root=\"1.2.826.0.1285.0.1.10\" extension=\"M85019\"/></agentOrganizationSDS>";
+//        final String control = "<device classCode=\"DEV\" determinerCode=\"INSTANCE\"><id root=\"1.2.826.0.1285.0.2.0.107\" extension=\"200000001613\"/></device>";
+//        final String test =    "<device classCode=\"DEV\" determinerCode=\"INSTANCE\"><id root=\"1.2.826.0.1285.0.2.0.107\" extension=\"200000000149\"/></device>";
+//        final String control = "<id root=\"DF91D420-DDC7-11ED-808B-AC162D1F16F0\"/>";
+//        final String test    = "<id root=\"DFBA6AC0-DDC7-11ED-808B-AC162D1F16F0\"/>";
+        final String control = "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\" ?><RCMR_IN030000UK06 xmlns=\"urn:hl7-org:v3\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xsi:schemaLocation=\"urn:hl7-org:v3 ..\\Schemas\\RCMR_IN030000UK06.xsd\"></RCMR_IN030000UK06>";
+        final String test    = "<?xml version=\"1.0\" encoding=\"UTF-8\" ?><RCMR_IN030000UK06 xmlns=\"urn:hl7-org:v3\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xsi:schemaLocation=\"urn:hl7-org:v3 ..\\Schemas\\RCMR_IN030000UK06.xsd\"></RCMR_IN030000UK06>";
+
+
+        Diff myDiff = DiffBuilder.compare(control).withTest(test)
+                .withNodeFilter(this::excludeComparisons)
+                .withDifferenceEvaluator(DifferenceEvaluators.chain(DifferenceEvaluators.Default,
+                        DifferenceEvaluators.downgradeDifferencesToEqual(ComparisonType.XML_STANDALONE)))
+                .checkForSimilar().build();
+
+        assertFalse(myDiff.toString(), myDiff.hasDifferences());
+    }
+
+    private static String getPayload(String gp2gpMessageBody) throws JSONException {
+        JSONObject jsonObject = new JSONObject(gp2gpMessageBody);
+        if (jsonObject.has("payload") ) {
+            return jsonObject.getString("payload");
+        } else {
+            return jsonObject.getJSONObject("request").getJSONObject("body").getString("payload");
+        }
+    }
```

Comment:
Could this too be put into a utility function in `src/test/java/uk/nhs/prm/deduction/e2e/utility/TestUtils.java`?